File: 199. Binary Tree Right Side View/solution.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None


class Solution:
    def rightSideView(self, root: TreeNode) -> List[int]:
        if root is None:
            return []
        que = []
        res = []
        que.append(root)
        while que:
            res.append(que[-1].val)
            length = len(que)
            for i in range(length):
                node = que.pop(0)
                if node:
                    if node.left is not None:
                        que.append(node.left)
                    if node.right is not None:
                        que.append(node.right)
        return res

File: 199. Binary Tree Right Side View/solution.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */


 func rightSideView(root *TreeNode) []int {
    if root == nil {
        return []int{}
    }
    que := []*TreeNode{root}
    res := []int{}
    for len(que) != 0 {
        res = append(res, que[len(que)-1].Val)
        
        length := len(que)
        for i := 0; i < length; i++ {
            node := que[0]
            que = que[1:]
            if node.Left != nil {
                que = append(que, node.Left)

            }
            if node.Right != nil {
                que = append(que, node.Right)
            }
        
        }

    }
    return res
}